### ISSUE TITLE
Rule Change: require-baseline might need to default to a warning

### ISSUE THREAD

**Issue by captainbrosset** [eslint/css#80]
### What rule do you want to change?

require-baseline

### What change do you want to make?

Generate fewer warnings

### How do you think the change should be implemented?

A new default behavior

### Example code

```css
body {
  accent-color: red;
}
```

### What does the rule currently do for this code?

` 18:3  error  Property 'accent-color' is not a widely available baseline feature  css/require-baseline`

### What will the rule do after it's changed?

` 18:3  warning Property 'accent-color' is not a widely available baseline feature  css/require-baseline`

### Participation

- [x] I am willing to submit a pull request to implement this change.

### Additional comments

Sorry, I haven't used ESLint in a long time, and don't remember if the default level of a rule can be changed, or if they are all errors by default, and can only be changed by users.

But, I think it would be better for the require-baseline rule to be a warning by default if possible.

The reason for this is that Baseline says:
> If this feature is Widely available, it's very safe to use it.

Baseline **does not say** the following:
> If this feature is not Widely available, don't use it.

There are many reasons why a developer might want to use a non-Widely available feature anyway. The feature might be a great candidate for progressive enhancement for example.

See https://github.com/web-platform-dx/web-features/issues/2758 for some context on this.

**Comment by captainbrosset** [comment]
Thank you for the clarification. I didn't remember this aspect of ESLint well.

So, yes, I agree, this issue comes down to whether the `recommended` config should set `require-baseline` as `error` or `warning`.

Based on the discussion in https://github.com/web-platform-dx/web-features/issues/2758, I think it should be a warning. Using the Baseline concept to recommend developers that they should **not** use a particular feature is not the intent of Baseline.

Let me ping a couple of Baseline folks from Google to share their opinions as well: @rachelandrew, @rviscomi, @atopal.

**Comment by nzakas** [comment]
@rachelandrew, @rviscomi, @atopal just following up from the previous comment. Do you have a suggestion for if the baseline rule would make more sense as a warning than an error?

**Comment by rviscomi** [comment]
> Based on the discussion in [web-platform-dx/web-features#2758](https://github.com/web-platform-dx/web-features/issues/2758), I think it should be a warning. Using the Baseline concept to recommend developers that they should **not** use a particular feature is not the intent of Baseline.

I don't personally see linter warnings as more fundamentally compatible with the Baseline philosophy than errors. In both cases, I would expect developers to take action to either remove the feature or suppress the linter. What separates the severity levels is the sense of urgency: forcing immediate action or deferring it until later. So I'm ok with keeping the recommended defaults but I wouldn't be strongly opposed to changing them.

**Comment by captainbrosset** [comment]
I agree that from the point of view of a linter, there isn't a huge difference between a warning or an error. Whether it's a warning or an error, a user would be expected to eventually address the linting message (by using a different property, or disabling the message).

The tension here, for me, comes from the fact that Baseline is not meant to say whether a feature should be used or not. Baseline isn't a YES/NO status. It's a YES/"HAVE YOU THOUGHT ABOUT IT AND DONE ENOUGH TESTING?" status.

What is not Baseline Widely Available is to be merely taken as a clue that more testing is needed.

This is why I worry about the default configuration here. If my gut instinct is right, and most users go with the defaults in whatever library they use, then most people will have the require-baseline rule configured as an error and set to Widely Available. And I worry that this will lead most users to not use Newly Available and Limited Availability features at all, which is not what Baseline is supposed to achieve.

**Comment by nzakas** [comment]
@captainbrosset it's possible you're attributing too much power to the default severity level of this rule. If your theory is that people will blindly use the default and obey it without thought, I can tell you from my experience with ESLint overall that this is rarely the case. The default configurations are there to set, pardon the pun, a baseline so people don't have to manually configure every rule individually. 

In practice what tends to happen is people start with the defaults, then something they want to do gets flagged as an error. So they investigate a bit and determine if they're wrong or if the rule is wrong. If they determine that the rule is wrong, then they can modify the configuration in a number of ways:

- They can turn the rule off completely
- They can change the rule to be a warning
- They can change the options of the rule
- They can disable the rule for just certain files
- They can disable the rule around just a segment of code using disable directives

ESLint has always been about end-user flexibility in how everything is configured. That's why we don't worry too much about the default options or severities. People will ultimately fine-tune their configuration to make it work the way they want.

**Comment by captainbrosset** [comment]
> it's possible you're attributing too much power to the default severity level of this rule. If your theory is that people will blindly use the default and obey it without thought, I can tell you from my experience with ESLint overall that this is rarely the case.

This was my working theory, yes. Thanks for sharing your experience. This is reassuring.

> ESLint has always been about end-user flexibility in how everything is configured.

I really appreciate this.

I would still be interested in hearing from @atopal and/or @rachelandrew on this question, as Baseline originally came from their team at Google Chrome. But I'm ok with @nzakas closing this if you want.

What I might do though is open a PR to make slight adjustments to the rule's documentation at https://github.com/eslint/css/blob/main/docs/rules/use-baseline.md

**Comment by nzakas** [comment]
We can leave this open to collect additional opinions. I'm not opposed to setting to a warning by default, I'd just like to see if there's some consensus over that before making the change.

**Comment by fasttime** [comment]
I think this was closed accidentally because of the commit message in ae5b544e14a97116a6bb4aae69eb6cbf389b501e. Reopening to collect feedback.

**Comment by nzakas** [comment]
After using the recommended config in the wild, I've come to agree that `use-baseline` would better be set at `warn` by default.
